### ISSUE TITLE
Syntax highlighter for nested block comments to be added

### ISSUE THREAD

**Issue by immadisairaj** [chapel-lang/chapel-tmbundle#19]
The highlighter should accept nested block comments.

```
/*
------something------
/*
------something------
*/
------something------
*/
```
The present situation is: For the above code only the first 5 lines are highlighted.
Expected: All the lines should be highlighted as comments.

**Comment by parthsarthiprasad** [comment]
This error is related to also tmbundle error , currently it after beginning of a nested block , searches for first instance of `*/`  , hence it closes the nested block at **first** instance , Need to restructure the tmbundle will fix the issue, I myself don't know the semantics but will try to do as soon as possible.

**Comment by parthsarthiprasad** [comment]
The current C++ version of highlighting semantics also work in the similar manner, The tmbundle searches for the first instance of */ , and closes the nested loop , Do you want something where the bundler searches for the last occourance of `*/` to close the nested loop.

**Nested loop in C working in similar fashion**
```C++
/*
------something------
/*
------something------
*/
------something------
*/
Where the last one is highlighted being a Stray `*/`
```

**Comment by immadisairaj** [comment]
> Here the compiler will search for last closing statement and comment out everything in between?

Yes, you are right if you want to go with that approach.

The other approach might be similar to checking for curly braces in a code. Which is nested
```
/* (1 open)
------something------
/* (2 open)
------something------
*/ (2 close)
------something------
*/ (1 close)
```

**Comment by lydia-duncan** [comment]
> Do you want something where the bundler searches for the last occourance of */ to close the nested loop

You definitely want to track the last *matching* `*/`.  @immadisairaj's nested strategy should be what you follow

**Comment by parthsarthiprasad** [comment]
> > Do you want something where the bundler searches for the last occourance of */ to close the nested loop
> 
> You definitely want to track the last _matching_ `*/`. @immadisairaj's nested strategy should be what you follow

I had a difference of opinion in this, Like all the nested comments are anyways going to be commented out , what will this change if we have nested multiline comment, If we have nesting or not, If we want to have a sense of sepration between blocks then maybe individually can use curly braces , 

Even though this can be done with the way we work with curly braces, will do the necessary changes then , with something like this.
https://github.com/chapel-lang/chapel-tmbundle/blob/822363c7e063444df0591ab559bf39cad2ebd44e/Syntaxes/Chapel.tmLanguage#L9-L15

**Comment by lydia-duncan** [comment]
> Like all the nested comments are anyways going to be commented out , what will this change if we have nested multiline comment

It depends on how you look for the last nested comment.  As you said, a program can easily have multiple multiline comments in it.  It would absolutely not be appropriate for the following to be treated as a single comment:

```chapel
/* blah blah */
proc someFunc() { ... }
/* blah blah 2 */
```

Going for the *first* closing comment after an opening comment is preferable to treating a whole file as commented.  But as this issue points out, the best strategy is to track matching opening and closing comments, ensuring that we only treat comments as comments and appropriately indicating when an opening comment is not matched by a closing one

**Comment by parthsarthiprasad** [comment]
> > Like all the nested comments are anyways going to be commented out , what will this change if we have nested multiline comment
> 
> It depends on how you look for the last nested comment. As you said, a program can easily have multiple multiline comments in it. It would absolutely not be appropriate for the following to be treated as a single comment:
> 
> ```chapel
> /* blah blah */
> proc someFunc() { ... }
> /* blah blah 2 */
> ```
> 
> Going for the _first_ closing comment after an opening comment is preferable to treating a whole file as commented. But as this issue points out, the best strategy is to track matching opening and closing comments, ensuring that we only treat comments as comments and appropriately indicating when an opening comment is not matched by a closing one

Yeah I checked few files and observed the requirement and this nested multiline comments might be a major requirement, will try to work for this as well then.
